### DISCUSSION TITLE
Apostrophe 3.11.0: The first release of 2022! 🐣

### DISCUSSION THREAD

**Discussion by agilbert** [apostrophecms/apostrophe#3608] [Announcements]
Happy New Year everyone!

We’re back at it with a new release for you this week that includes the migration of our `apostrophe-passport` module to A3. The new [Passport Bridge](https://github.com/apostrophecms/passport-bridge) module will give you the ability to authenticate your CMS users against popular services like Google and Okta, with all of the flexibility and extensibility that [Passport.js](https://www.passportjs.org/) provides.

We've also published some new [documentation](https://v3.docs.apostrophecms.org/) that we're excited to share. This will be especially valuable for any developers looking to extend or modify the core UI of A3.

* [Customizing the user interface](https://v3.docs.apostrophecms.org/guide/custom-ui.html) will guide you through the steps to customize the admin UI in Apostrophe.
* [Custom schema field types](https://v3.docs.apostrophecms.org/guide/custom-schema-field-types.html) will explain how to create new field types for your documents.
* [Querying the database](https://v3.docs.apostrophecms.org/guide/database-queries.html) will help you fetch all manner of data directly from MongoDB for easy use right in your own modules and templates.

Other than that, it’s a relatively quiet release for us, but we’re just getting started for the year. Stay tuned for more updates to our migration utilities, as well as new stability, performance, and feature improvements in the coming weeks and months.

As always we welcome you to check out [our public roadmap](https://portal.productboard.com/apostrophecms/1-product-roadmap) and share any feedback with us there. Or find us in [Discord](http://chat.apostrophecms.com/) or here in the GitHub discussions if you want to get a different kind of conversation going.

Read on below for the full release notes and here’s to a great year ahead 💪 

### Adds

* Apostrophe now extends Passport's `req.login` to emit an `afterSessionLogin` event from the `@apostrophecms:login` module, with `req` as an argument. Note that this does not occur at all for login API calls that return a bearer token rather than establishing an Express session.

### Fixes

* Apostrophe's extension of `req.login` now accounts for the `req.logIn` alias and the skippable `options` parameter, which is relied upon in some `passport` strategies.

### Changes

* Unpins `vue-click-outside-element` the packaging of which has been fixed upstream.
* Adds deprecation note to `__testDefaults` option. It is not in use, but removing would be a minor BC break we don't need to make.
* Allows test modules to use a custom port as an option on the `@apostrophecms/express` module.
* Removes the code base pull request template to instead inherit the organization-level template.

## Apostrophe 3.x modules

[passport-bridge 1.0.0-beta](https://github.com/apostrophecms/passport-bridge)

* Initial beta release. Provides a single sign-on bridge to Google, Twitter, Facebook, GitLab and many other identity providers.

palette 3.0.0

* Stable release. Compatible with the @apostrophecms/security-headers module. Available as part of [Apostrophe Assembly](https://apostrophecms.com/assembly).

**Comment by ytilis** [comment]
Not sure if anyone's pointed this out yet but the `passport-bridge` module you're linking to seems to still be private. Clicking the link to it in this thread just throws a 404 for me.

**Comment by boutell** [comment]
Thank you for telling us. The module is public in npm, but the repository was inadvertently private. It is now public.
